exampleCourse/questions/demo/autograder/python/fillBlankLastElement/question.html

```diff
@@ -0,0 +1,41 @@
+<pl-question-panel>
+    <p>
+        <i>
+            This question demonstrates the creation of a file from other elements in the question. It uses a "fill-in-the-blank" model to allow students to write code elements that will be combined into a single file.
+        </i>
+    </p>
+
+    <p>
+        Complete the following function that finds the last element of a list and returns it. The function should be named <code>last_element</code> and should take a single argument, which is the list from which to extract the last element.
+    </p>
+
+    <pl-external-grader-variables params-name="names_for_user" empty="true">
+    </pl-external-grader-variables>
+
+    <p>Your code snippet should define the following functions:</p>
+    <pl-external-grader-variables params-name="names_from_user">
+        <pl-variable name="last_element" type="function">Returns the last element of a list</pl-variable>
+    </pl-external-grader-variables>
+    <pl-overlay width="600" height="200" clip="false">
+      <pl-background>
+<pl-code language="python">
+def last_element(lst: list):
+    """    Returns the last element of a list.
+    Args:
+        lst (list): The list from which to extract the last element.
+    Returns:
+        The last element of the list.
+    """
+
+    return lst[                                      ]
+</pl-code></pl-background>
+      <pl-location left="270" top="155">
+        <pl-string-input answers-name="index" display="inline" size="30" show-help-text="false"></pl-string-input>
+      </pl-location>
```

Comment:
The use of overlay here is a bit of a hack, I welcome any better suggestions.

Reply:
Is there any way to have `pl-code` ignore `pl-*` inputs for highlighting? I see:

```
code = pl.inner_html(element).removeprefix("\r").removeprefix("\n")
```

it would be a nice feature to directly embed inputs on the element.

Reply:
I think limiting it to `pl-` element might be too strict, maybe a question author wants to use an `<input...>` or `<span>` inside the element.

One option is to have an attribute in `pl-code` with something like `escape-html="false"` (default true), which would accept HTML tags inside and let the question author take care of escaping anything that should be escaped. This would be tricky, though, if we want to combine this with syntax highlighting. I'll leave it as future work, feel free to add an issue if you believe this is useful.

Reply:
Without that ability (https://github.com/PrairieLearn/PrairieLearn/issues/12294), I think that an overlay is a good way to do this, I can't think of another way.